src/CommonITILValidation.php

```diff
@@ -392,9 +393,18 @@ public function post_addItem()
 
     public function prepareInputForUpdate($input)
     {
-
+        $can_answer = static::canValidate($this->fields[static::$items_id], $this->getID());
         $forbid_fields = ['entities_id', static::$items_id, 'is_recursive'];
-        if (isset($input["status"]) && static::canValidate($this->fields[static::$items_id])) {
+
+        if (!$can_answer) {
+            array_push($forbid_fields, 'status', 'comment_validation', 'validation_date');
+        }
+
+        foreach ($forbid_fields as $key) {
+            unset($input[$key]);
+        }
+
+        if (isset($input["status"])) {
```

Comment:
It seems that there is not anymore any protection against a change of `users_id`/`itemtype_target`/`items_id_target` fields by a user that is not supposed to be able to create validation requests.

It means that anyone being the target of a validation could alter the validation request creator/target, by injecting the corresponding values in the HTTP request.

1. `users_id` field should probably be protected against changes.
2. `itemtype_target`/`items_id_target` should be protected against changes as long as the current connected user is not matching the `users_id` field.

Some additional tests should be added to validate this.

Reply:
The requesting user, and target of the approval should probably be readonly for everyone after the approval was created. If an approval was sent for the wrong person/group, the approval should be removed and recreated.